### PR TITLE
cloudtest: Cross-repo coordination for cloud tests

### PR DESCRIPTION
**Pull request by necaris**
Update the pull request template to reference cloud tests as well as
orchestration, and add the Cloud team as CODEOWNERS for the shared cloud
test code.

### Motivation

   * This PR refactors existing code.

   Not code so much as process :smile:

### CODE REVIEW

File: .github/CODEOWNERS
Comment (nrainer-materialize) [inline .github/CODEOWNERS:97]: I would like to wait with this change until we actually move the code and then restrict the path to the specific subfolder that will contain the shared logic. In particular, I would not want this ownership for cloud _test cases_ that reside in the mz repo.

Furthermore, I would like @philip-stoev to sign this off.

**Review by nrainer-materialize — CHANGES_REQUESTED** [review]
Thank you for the preparation of this change!

File: .github/CODEOWNERS
Comment (benesch) [reply .github/CODEOWNERS:97]: Is there a quick summary of what functions are going to be owned by the cloud team? Would it make sense to consider shared ownership between the cloud and QA teams for those functions?

Comment (nrainer-materialize) [reply .github/CODEOWNERS:97]: Yeah, I envision a shared ownership, so we should add the QA team as well if it is not listed yet.

The exact set of functions is still to be defined but will roughly be [tests/k8s/util (currently in cloud repo)](https://github.com/MaterializeInc/cloud/tree/main/tests/k8s/util).

Comment (philip-stoev) [reply .github/CODEOWNERS:97]: yes, I think the QA team should also be listed here (unless already inherited from some higher-level CODEOWNERS rule).

**Review by philip-stoev — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
Please don't merge this yet.

**Review by nrainer-materialize — APPROVED** [review]

**Pull request merged**
